refactor for best practices

newellista opened the pull request:
Following the direction of @mmmries, I made a few changes

Commit:
refactor for best practices

Commit:
fix errors

newellista commented on lib/weather/weather_report.ex:
```diff
@@ -32,12 +31,9 @@ defmodule Weather.WeatherReport do
   def parse_xml({xml, _}) do
-    Enum.into(@needed_fields, %{}, fn node_name -> extract_node(xml, node_name) end)
+    {:ok, Enum.into(@needed_fields, %{}, fn node_name -> extract_node(xml, node_name) end) }
   end
```
I'm not sure if this is the best/right way to do this.  Any input welcome!

mmmries replied:
What about `Enum.into(@needed_fields, %{}, &( extract_node(xml, &1) ))`?

Commit:
Refactor `parse_xml`

`parse_xml` was originally returning a tuple, but, I realized that I
didn't need to do this, so this refactor simplifies things.

Commit:
refactor parse_xml

mmmries commented on lib/weather/weather_report.ex:
```diff
@@ -32,12 +31,9 @@ defmodule Weather.WeatherReport do
+
   def parse_xml({xml, _}) do
-    Enum.into(@needed_fields, %{}, fn node_name -> extract_node(xml, node_name) end)
+    Enum.into(@needed_fields, %{}, &(extract_node(xml, &1)) )
```
what did you end up thinking of this? do you prefer the shortcut syntax for anonymous functions or the longer form?

newellista replied:
I think that right now, the longer form is more discoverable.  However, since the short form is more idiomatic, that's what I'm trying to go with.

My real question around this line regarded the tuple I had wrapped things in.  I was confused about how to do this, since the other instances of `handle_response` returned tuples, with a status code as the first member.

I had an epiphany last night, wherein I remembered that I could have different method signatures for `print_response`.  I deleted some code, the remaining code was simpler, and it was much more readable.
